### ISSUE TITLE
CA1416 errors when using WinForms and .NET 8

### ISSUE THREAD

**Issue by SoftCircuits** [dotnet/roslyn-analyzers#7243]
I have a WinForms application that has been working fine with .NET 6. But today I've updated it to .NET 8.

```
<PropertyGroup>
  <OutputType>WinExe</OutputType>
  <TargetFramework>net8.0-windows</TargetFramework>
  <UseWindowsForms>true</UseWindowsForms>
  <Nullable>enable</Nullable>
  <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
</PropertyGroup>
```

Now, I suddenly get hundreds of warnings, such as the following.

```
3>C:\Users\jwood\Source\Repos\SoftCircuits\HtmlMonkey\TestHtmlParse\HtmlVisualizer.cs(92,21,92,40): warning CA1416: This call site is reachable on all platforms. 'ListView.Items' is only supported on: 'windows' 6.1 and later. (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1416)
3>C:\Users\jwood\Source\Repos\SoftCircuits\HtmlMonkey\TestHtmlParse\NodeVisualizer.cs(157,30,157,62): warning CA1416: This call site is reachable on all platforms. 'ListView.ColumnHeaderCollection.Add(string?)' is only supported on: 'windows' 6.1 and later. (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1416)
3>C:\Users\jwood\Source\Repos\SoftCircuits\HtmlMonkey\TestHtmlParse\NodeVisualizer.cs(154,13,154,27): warning CA1416: This call site is reachable on all platforms. 'ListView.Items' is only supported on: 'windows' 6.1 and later. (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1416)
3>C:\Users\jwood\Source\Repos\SoftCircuits\HtmlMonkey\TestHtmlParse\HtmlVisualizer.cs(86,21,86,33): warning CA1416: This call site is reachable on all platforms. 'TextBox.Text' is only supported on: 'windows' 6.1 and later. (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1416)
3>C:\Users\jwood\Source\Repos\SoftCircuits\HtmlMonkey\TestHtmlParse\NodeVisualizer.cs(157,30,157,46): warning CA1416: This call site is reachable on all platforms. 'ListView.Columns' is only supported on: 'windows' 6.1 and later. (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1416)
3>C:\Users\jwood\Source\Repos\SoftCircuits\HtmlMonkey\TestHtmlParse\NodeVisualizer.cs(158,17,158,29): warning CA1416: This call site is reachable on all platforms. 'ColumnHeader.Width' is only supported on: 'windows' 6.1 and later. (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1416)
```

This appears to be related somehow to *assemblyinfo.cs*, and syntax along the lines of `[assembly: SupportedOSPlatform("windows")]`.

Why am I getting these warnings on .NET 8, but not .NET 6 or .NET 7? Did .NET 8 something change?

What is the fix? I found a suggestion to set `<GenerateAssemblyInfo>true</GenerateAssemblyInfo>`, but when I tried this, I got a bunch of CS0579 errors telling me I have duplicate attributes defined.

**Comment by buyaa-n** [comment]
> Why am I getting these warnings on .NET 8, but not .NET 6 or .NET 7? Did .NET 8 something change?

Nothing changed in .NET 8 from analyzer side, this issue might be caused by SDK or WinForms change, might want a file an issue there. 

> What is the fix? I found a suggestion to set <GenerateAssemblyInfo>true</GenerateAssemblyInfo>, but when I tried this, I got a bunch of CS0579 errors telling me I have duplicate attributes defined.

Setting `GenerateAssemblyInfo` into true would cause `AssemblyInfo.cs` file generated by MSBuild and `[assembly: SupportedOSPlatform("windows")]` and other attributes added to it, you might have had a custom `AssemblyInfo.cs` and therefore some attributes are duplicated

**Comment by SoftCircuits** [comment]
@buyaa-n Thanks. Unfortunately, the winforms repository has very restrictive issue reporting, and doesn't seem to have a category for an issue like this.

**Comment by buyaa-n** [comment]
If you have custom `AssemblyInfo.cs`, and if it has no `[assembly: SupportedOSPlatform("windows")]` attribute, adding the attribute to it and disabling the `GenerateAssemblyInfo` might help

**Comment by SoftCircuits** [comment]
@buyaa-n That does appear to solve it, yes. Although I really don't understand why it was needed or why it wasn't needed with .NET 6 or 7. Thanks!

**Comment by buyaa-n** [comment]
The analyzer, and attribute handling have been like this since .NET6, nothing changed in 8.0 closing the issue by design. Maybe WinForms annotated their library as windows only recently.

**Comment by SoftCircuits** [comment]
@buyaa-n And a year later, I'm hitting the same issue again. I set `GenerateAssemblyInfo` to true, but I still get dozens of warnings.

This project previously built without error. Why should it be so hard to write for Windows now?

This was closed as "not planned." I wonder if someone can clarify what is not planned? Something is breaking working code. Shouldn't there be some effort to make this transition a little easier than it is?
